src/bindings/g_object/object.cr

```diff
@@ -508,6 +508,12 @@ module GObject
         end
 
         def emit({{ signature.args.splat }})
+          {% for arg in signature.args %}
+            {% resolved_type = arg.type.resolve %}
+            {% if resolved_type < Reference && !resolved_type.has_method?(:to_unsafe) %}
+              {{ arg.var }} = Pointer(Void).new({{ arg.var }}.object_id)
```

Comment:
Here you can just use `{{ arg.var }} = {{ arg.var }}.as(Void*)`.
I think that's a bit clearer.

Reply:
hmmm, nice, I didn't know this cast would work.